core/src/test/java/com/linecorp/armeria/common/stream/StreamWriterOutputStreamTest.java

```diff
@@ -0,0 +1,311 @@
+        final OutputStream outputStream = writer.toOutputStream(x -> {
+            final byte[] data = new byte[x.byteBuf().readableBytes()];
+            x.byteBuf().readBytes(data);
+            return new String(data);
+        }, 3);
+
+        for (byte b : "foobarbaz".getBytes()) {
+            outputStream.write(b);
+        }
+        outputStream.close();
+
+        StepVerifier.create(writer)
+                    .expectNext("foo", "bar", "baz")
+                    .verifyComplete();
+    }
+
+    @Test
+    void writeWithOffset_exceedMaxBufferSize() throws IOException {
+        final DefaultStreamMessage<String> writer = new DefaultStreamMessage<>();
+        final OutputStream outputStream = writer.toOutputStream(x -> {
+            final byte[] data = new byte[x.byteBuf().readableBytes()];
+            x.byteBuf().readBytes(data);
+            return new String(data);
+        }, 3);
+        final List<String> strings = ImmutableList.of("3", "456", "789", "ABC", "D");
+        final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        byteArrayOutputStream.write(0);
+        byteArrayOutputStream.write(strings.stream()
+                                           .map(String::getBytes)
+                                           .reduce(Bytes::concat).get());
+        byteArrayOutputStream.write(0);
+
+        for (byte b : "12".getBytes()) {
+            outputStream.write(b);
+        }
+        outputStream.write(byteArrayOutputStream.toByteArray(), 1, byteArrayOutputStream.size() - 2);
+        outputStream.close();
+
+        StepVerifier.create(writer)
+                    .expectNext("123", "456", "789", "ABC", "D")
+                    .verifyComplete();
+    }
+
+    @ValueSource(ints = { Integer.MIN_VALUE, -1, 0 })
+    @ParameterizedTest
+    void write_nonPositiveMaxBufferSize(int maxBufferSize) throws IOException {
+        final DefaultStreamMessage<String> writer = new DefaultStreamMessage<>();
+        assertThatThrownBy(() -> writer.toOutputStream(x -> {
+            final byte[] data = new byte[x.byteBuf().readableBytes()];
+            x.byteBuf().readBytes(data);
+            return new String(data);
+        }, maxBufferSize))
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessage("maxBufferSize should be positive");
+    }
+
+    @Test
+    void close() throws IOException {
+        final DefaultStreamMessage<Integer> writer = new DefaultStreamMessage<>();
+        final OutputStream outputStream = writer.toOutputStream(x -> Integer.valueOf(x.byteBuf().readByte()),
+                                                                128);
+
+        final List<Integer> integers = ImmutableList.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+        for (Integer integer : integers) {
+            outputStream.write(integer.byteValue());
+            outputStream.flush();
+        }
+
+        assertDoesNotThrow(outputStream::close);
```

Comment:
nit: We prefer using AssertJ if possible.
```suggestion
        assertThatCode(outputStream::close).doesNotThrowAnyException();
```

Reply:
✅ updated!